src/goto-programs/resolve_concrete_function_call.cpp

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************\
+
+ Module: Goto Program utilties
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <algorithm>
+
+#include "resolve_concrete_function_call.h"
+
+/// See the operator() method comment.
+/// \param symbol_table: The symbol table to resolve the function call in
+resolve_concrete_function_callt::resolve_concrete_function_callt(
+  const symbol_tablet &symbol_table):
+    symbol_table(symbol_table)
+{
+  class_hierarchy(symbol_table);
```

Comment:
I'd recommend taking a const ref to the class hierarchy, as users are likely to have need of it themselves and it would be good to avoid rebuilding it unnecessarily.